query-node/mappings/src/content/utils.ts

```diff
@@ -0,0 +1,484 @@
+  }
+
+  const rawAsset = assets[assetIndex]
+
+  // escape if asset is describing URLs (can't get size)
+  if (rawAsset.isUrls) {
+    return undefined
+  }
+
+  // !rawAsset.isUrls && rawAsset.isUpload // asset is in storage
+
+  // extract video size
+  const contentParameters: ContentParameters = rawAsset.asUpload
+  // `size` is masked by `size` special name in struct that's why there needs to be `.get('size') as u64`
+  const videoSize = (contentParameters.get('size') as unknown as u64).toBn()
+
+  return videoSize
+}
+
+async function prepareLanguage(languageIso: string, db: DatabaseManager, blockNumber: number): Promise<Language> {
+  // validate language string
+  const isValidIso = ISO6391.validate(languageIso);
+
+  // ensure language string is valid
+  if (!isValidIso) {
+    return inconsistentState(`Invalid language ISO-639-1 provided`, languageIso)
+  }
+
+  // load language
+  const language = await db.get(Language, { where: { iso: languageIso } as FindConditions<Language> })
+
+  // return existing language if any
+  if (language) {
+    return language;
+  }
+
+
+  // create new language
+  const newLanguage = new Language({
+    iso: languageIso,
+    createdInBlock: blockNumber,
+
+    // TODO: remove these lines after Hydra auto-fills the values when cascading save (remove them on all places)
+    createdById: '1',
+    updatedById: '1',
+  })
+
+  await db.save<Language>(newLanguage)
+
+  return newLanguage
+}
+
+async function prepareLicense(licenseProtobuf: LicenseMetadata.AsObject): Promise<License> {
+  // NOTE: Deletion of any previous license should take place in appropriate event handling function
+  //       and not here even it might appear so.
+
+  // crete new license
+  const license = new License({
+    ...licenseProtobuf,
+
+    createdById: '1',
+    updatedById: '1',
+  })
+
+  return license
+}
+
+async function prepareVideoMetadata(videoProtobuf: VideoMetadata.AsObject, videoSize: BN | undefined): Promise<VideoMediaMetadata> {
+  // create new encoding info
+  const encoding = new VideoMediaEncoding(videoProtobuf.mediaType)
+
+  // create new video metadata
+  const videoMeta = new VideoMediaMetadata({
+    encoding,
+    pixelWidth: videoProtobuf.mediaPixelWidth,
+    pixelHeight: videoProtobuf.mediaPixelHeight,
+
+    createdById: '1',
+    updatedById: '1',
+  })
+
+  // fill in video size if provided
+  if (videoSize !== undefined) {
+    videoMeta.size = videoSize
+  }
+
+  return videoMeta
+}
+
+async function prepareVideoCategory(categoryId: number, db: DatabaseManager): Promise<VideoCategory> {
+  // load video category
+  const category = await db.get(VideoCategory, { where: { id: categoryId.toString() } as FindConditions<VideoCategory> })
+
+  // ensure video category exists
+  if (!category) {
+    return inconsistentState('Non-existing video category association with video requested', categoryId)
```

Comment:
Mistake in metadata, ie user selecting no-existant category should not halt the processor.

Reply:
Invalid state doesn't throw error anymore.